lizmap/server/lizmap_accesscontrol.py

```diff
@@ -0,0 +1,284 @@
+                    # Reset edition rights
+                    rights.canInsert = rights.canUpdate = rights.canDelete = False
+            else:
+                # The layer has no editionLayers config defined
+                # Reset edition rights
+                QgsMessageLog.logMessage("No edition config defined for layer: %s (%s)" % (layer_name, layer_id), "lizmap", Qgis.Info)
+                rights.canInsert = rights.canUpdate = rights.canDelete = False
+        else:
+            # No editionLayers defined
+            # Reset edition rights
+            QgsMessageLog.logMessage("Lizmap config has no editionLayers", "lizmap", Qgis.Info)
+            rights.canInsert = rights.canUpdate = rights.canDelete = False
+
+        # Check Lizmap layer config
+        if layer_name not in cfg_layers or not cfg_layers[layer_name]:
+            # Lizmap layer config not defined
+            QgsMessageLog.logMessage("Lizmap config has no layer: %s" % layer_name, "lizmap", Qgis.Warning)
+            # Default layer rights applied
+            return rights
+
+        # Check Lizmap layer group visibility
+        cfg_layer = cfg_layers[layer_name]
+        if 'group_visibility' not in cfg_layer or not cfg_layer['group_visibility']:
+            # Lizmap config has no options
+            QgsMessageLog.logMessage("No Lizmap layer group visibility for: %s" % layer_name, "lizmap", Qgis.Info)
+            # Default layer rights applied
+            return rights
+
+        # Get Lizmap layer group visibility
+        group_visibility = cfg_layer['group_visibility'].split(',')
+        group_visibility = [g.strip() for g in group_visibility]
+
+        # If one Lizmap user group provided in request headers is
+        # defined in Lizmap layer group visibility, the default layer
+        # rights is applied
+        for g in groups:
+            if g in group_visibility:
+                QgsMessageLog.logMessage(
+                    "Group %s is in Lizmap layer group visibility for: %s" % (g, layer_name),
+                    "lizmap", Qgis.Info)
+                return rights
+
+        # The lizmap user groups provided gy the request are not
+        # authorized to get access to the layer
+        QgsMessageLog.logMessage(
+            "Groups %s is in Lizmap layer group visibility for: %s" % (', '.join(groups), layer_name),
+            "lizmap", Qgis.Info)
+        rights.canRead = False
+        rights.canInsert = rights.canUpdate = rights.canDelete = False
+        return rights
+
+    def authorizedLayerAttributes(self, layer: 'QgsVectorLayer', attributes: 'Iterable[str]') -> 'List[str]':
+        """ Return the authorised layer attributes """
+        return super().authorizedLayerAttributes(layer, attributes)
+
+    def allowToEdit(self, layer: 'QgsVectorLayer', feature: 'QgsFeature') -> bool:
+        """ Are we authorise to modify the following geometry """
+        return super().allowToEdit(layer, feature)
+
```

Comment:
```suggestion
```
Same as above?

Reply:
We don't have yet a way to configured, but I like to get it to think about it.